backend/src/uploads/s3Service.ts

```diff
@@ -40,13 +40,7 @@ export const s3Service = (config: S3Config, prefix: string) => {
       location = `https://${location}`
     }
 
-    if (!S3_PUBLIC_GATEWAY) {
-      return location
-    }
-
-    const publicLocation = new URL(S3_PUBLIC_GATEWAY)
-    publicLocation.pathname = new URL(location).pathname
-    return publicLocation.href
```

Comment:
Let's not write that to the database but replace the domain host on-the-fly during a request.

Reply:
We're lucky this environment value didn't make it on production yet.